Stracktrace with scm scanning

`--url` takes a URL (obviously), but it would be nice if the tool didn't throw an exception if given something like `git@github.com:strimzi/strimzi-kafka-oauth.git` which would be acceptable to `git clone` but is not a URL. An error message to say "use a URL" would suffice, but it would be even nicer if it accepted that.

```
$ java -jar target/srcclr-1.4-SNAPSHOT-jar-with-dependencies.jar --product='cpe:/a:redhat:amq_streams:1.3' --product-version='1.3.0.CR1' scm --url git@github.com:strimzi/strimzi-kafka-oauth.git --ref master
INFO com.redhat.engineering.srcclr.SCM- Scanning local file system with git@github.com:strimzi/strimzi-kafka-oauth.git
INFO c.r.engineering.srcclr.SrcClrInvoker- Invoking in environment {GIT_BRANCH=null, GIT_URL=null, GOPATH=/tmp/sourceclear-gopath-218570056154285958} command [/opt/srcclr-3.4.3/jre/bin/java, -jar, /opt/srcclr-3.4.3/srcclr-3.4.3.jar, scan, --json, --skip-collectors="SOFile", git@github.com:strimzi/strimzi-kafka-oauth.git, --ref, master] ....
ERROR c.r.engineering.srcclr.SrcClrInvoker- 
ERROR c.r.engineering.srcclr.SrcClrInvoker- Invalid exit 
org.zeroturnaround.exec.InvalidExitValueException: Unexpected exit value: 1, allowed exit values: [0], executed command [/opt/srcclr-3.4.3/jre/bin/java, -jar, /opt/srcclr-3.4.3/srcclr-3.4.3.jar, scan, --json, --skip-collectors="SOFile", git@github.com:strimzi/strimzi-kafka-oauth.git, --ref, master] in directory /tmp/sourceclear-invoker-8487097337571164151 with environment {GIT_BRANCH=null, GIT_URL=null, GOPATH=/tmp/sourceclear-gopath-218570056154285958}, output was 0 bytes:

	at org.zeroturnaround.exec.InvalidExitUtil.checkExit(InvalidExitUtil.java:27)
	at org.zeroturnaround.exec.WaitForProcess.call(WaitForProcess.java:114)
	at org.zeroturnaround.exec.ProcessExecutor.waitFor(ProcessExecutor.java:1097)
	at org.zeroturnaround.exec.ProcessExecutor.execute(ProcessExecutor.java:925)
	at com.redhat.engineering.srcclr.SrcClrInvoker.execSourceClear(SrcClrInvoker.java:138)
	at com.redhat.engineering.srcclr.SCM.call(SCM.java:119)
	at com.redhat.engineering.srcclr.SCM.call(SCM.java:40)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1743)
	at picocli.CommandLine.access$900(CommandLine.java:145)
	at picocli.CommandLine$RunAll.handle(CommandLine.java:2157)
	at picocli.CommandLine$RunAll.handle(CommandLine.java:2116)
	at picocli.CommandLine$AbstractParseResultHandler.handleParseResult(CommandLine.java:1928)
	at picocli.CommandLine.parseWithHandlers(CommandLine.java:2282)
	at com.redhat.engineering.srcclr.SrcClrWrapper.invokeWrapper(SrcClrWrapper.java:123)
	at com.redhat.engineering.srcclr.SrcClrWrapper.main(SrcClrWrapper.java:111)
Exception in thread "main" com.redhat.engineering.srcclr.utils.InternalException: Error executing SourceClear 
	at com.redhat.engineering.srcclr.SrcClrInvoker.execSourceClear(SrcClrInvoker.java:164)
	at com.redhat.engineering.srcclr.SCM.call(SCM.java:119)
	at com.redhat.engineering.srcclr.SCM.call(SCM.java:40)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1743)
	at picocli.CommandLine.access$900(CommandLine.java:145)
	at picocli.CommandLine$RunAll.handle(CommandLine.java:2157)
	at picocli.CommandLine$RunAll.handle(CommandLine.java:2116)
	at picocli.CommandLine$AbstractParseResultHandler.handleParseResult(CommandLine.java:1928)
	at picocli.CommandLine.parseWithHandlers(CommandLine.java:2282)
	at com.redhat.engineering.srcclr.SrcClrWrapper.invokeWrapper(SrcClrWrapper.java:123)
	at com.redhat.engineering.srcclr.SrcClrWrapper.main(SrcClrWrapper.java:111)
Caused by: org.zeroturnaround.exec.InvalidExitValueException: Unexpected exit value: 1, allowed exit values: [0], executed command [/opt/srcclr-3.4.3/jre/bin/java, -jar, /opt/srcclr-3.4.3/srcclr-3.4.3.jar, scan, --json, --skip-collectors="SOFile", git@github.com:strimzi/strimzi-kafka-oauth.git, --ref, master] in directory /tmp/sourceclear-invoker-8487097337571164151 with environment {GIT_BRANCH=null, GIT_URL=null, GOPATH=/tmp/sourceclear-gopath-218570056154285958}, output was 0 bytes:

	at org.zeroturnaround.exec.InvalidExitUtil.checkExit(InvalidExitUtil.java:27)
	at org.zeroturnaround.exec.WaitForProcess.call(WaitForProcess.java:114)
	at org.zeroturnaround.exec.ProcessExecutor.waitFor(ProcessExecutor.java:1097)
	at org.zeroturnaround.exec.ProcessExecutor.execute(ProcessExecutor.java:925)
	at com.redhat.engineering.srcclr.SrcClrInvoker.execSourceClear(SrcClrInvoker.java:138)
	... 10 more
```

The url is passed directly through to srrclr and it doesn't accept it. However I do agree the amount of stacktrace could be improved. In my dev version I have changed the output so the sourceclear error is visible and reduced the amount of stacktrace.

**INFO**
```
 java -jar target/srcclr-1.4-SNAPSHOT-jar-with-dependencies.jar --processor=cve --product=test --product-version=test scm --url="git@github.com:strimzi/strimzi-kafka-oauth.git" |& tee ~/tmp/build.log 
INFO c.r.engineering.srcclr.SrcClrInvoker- Invoking in environment {GIT_BRANCH=null, GIT_URL=null, GOPATH=/tmp/sourceclear-gopath-8150033951073120558} command [/opt/srcclr-3.4.7/jre/bin/java, -jar, /opt/srcclr-3.4.7/srcclr-3.4.7.jar, scan, --json, --skip-collectors="SOFile", --url, git@github.com:strimzi/strimzi-kafka-oauth.git] ....
ERROR c.r.engineering.srcclr.SrcClrInvoker- 
ERROR c.r.engineering.srcclr.SrcClrInvoker- Couldn't clone git@github.com:strimzi/strimzi-kafka-oauth.git, this URL appears invalid.
ERROR c.r.engineering.srcclr.SrcClrInvoker- 
ERROR c.r.engineering.srcclr.SrcClrInvoker- Invalid exit: Unexpected exit value: 1, allowed exit values: [0], executed command [/opt/srcclr-3.4.7/jre/bin/java, -jar, /opt/srcclr-3.4.7/srcclr-3.4.7.jar, scan, --json, --skip-collectors="SOFile", --url, git@github.com:strimzi/strimzi-kafka-oauth.git] in directory /tmp/sourceclear-invoker-7738077293840475912 with environment {GIT_BRANCH=null, GIT_URL=null, GOPATH=/tmp/sourceclear-gopath-8150033951073120558}, output was 0 bytes:
```


**DEBUG**
```
java -jar target/srcclr-1.4-SNAPSHOT-jar-with-dependencies.jar --processor=cve --product=test --product-version=test scm --url="git@github.com:strimzi/strimzi-kafka-oauth.git" -d |& tee ~/tmp/build.log
DEBUG c.r.engineering.srcclr.SrcClrWrapper- SourceClearWrapper 1.4-SNAPSHOT ( SHA : ea2e5230)
DEBUG c.r.engineering.srcclr.SrcClrWrapper- Read configuration from /home/rnc/.srcclr/invoker.properties with contents {email-server=smtp.redhat.com, processor=cve}
INFO c.r.engineering.srcclr.SrcClrInvoker- Invoking in environment {GIT_BRANCH=null, GIT_URL=null, GOPATH=/tmp/sourceclear-gopath-3060457657426141852} command [/opt/srcclr-3.4.7/jre/bin/java, -jar, /opt/srcclr-3.4.7/srcclr-3.4.7.jar, scan, --json, --skip-collectors="SOFile", --url, git@github.com:strimzi/strimzi-kafka-oauth.git] ....
DEBUG o.z.exec.ProcessExecutor- Executing [/opt/srcclr-3.4.7/jre/bin/java, -jar, /opt/srcclr-3.4.7/srcclr-3.4.7.jar, scan, --json, --skip-collectors="SOFile", --url, git@github.com:strimzi/strimzi-kafka-oauth.git] in /tmp/sourceclear-invoker-7566823331311174600 with environment {GIT_BRANCH=null, GIT_URL=null, GOPATH=/tmp/sourceclear-gopath-3060457657426141852}.
DEBUG o.z.exec.ProcessExecutor- Started java.lang.UNIXProcess@167fdd33
ERROR c.r.engineering.srcclr.SrcClrInvoker- 
ERROR c.r.engineering.srcclr.SrcClrInvoker- Couldn't clone git@github.com:strimzi/strimzi-kafka-oauth.git, this URL appears invalid.
DEBUG o.zeroturnaround.exec.WaitForProcess- java.lang.UNIXProcess@167fdd33 stopped with exit code 1
ERROR c.r.engineering.srcclr.SrcClrInvoker- 
ERROR c.r.engineering.srcclr.SrcClrInvoker- Invalid exit: Unexpected exit value: 1, allowed exit values: [0], executed command [/opt/srcclr-3.4.7/jre/bin/java, -jar, /opt/srcclr-3.4.7/srcclr-3.4.7.jar, scan, --json, --skip-collectors="SOFile", --url, git@github.com:strimzi/strimzi-kafka-oauth.git] in directory /tmp/sourceclear-invoker-7566823331311174600 with environment {GIT_BRANCH=null, GIT_URL=null, GOPATH=/tmp/sourceclear-gopath-3060457657426141852}, output was 0 bytes:

DEBUG c.r.engineering.srcclr.SrcClrInvoker- SourceClear process finished with an error 
org.zeroturnaround.exec.InvalidExitValueException: Unexpected exit value: 1, allowed exit values: [0], executed command [/opt/srcclr-3.4.7/jre/bin/java, -jar, /opt/srcclr-3.4.7/srcclr-3.4.7.jar, scan, --json, --skip-collectors="SOFile", --url, git@github.com:strimzi/strimzi-kafka-oauth.git] in directory /tmp/sourceclear-invoker-7566823331311174600 with environment {GIT_BRANCH=null, GIT_URL=null, GOPATH=/tmp/sourceclear-gopath-3060457657426141852}, output was 0 bytes:

        at org.zeroturnaround.exec.InvalidExitUtil.checkExit(InvalidExitUtil.java:27)
        at org.zeroturnaround.exec.WaitForProcess.call(WaitForProcess.java:114)
        at org.zeroturnaround.exec.ProcessExecutor.waitFor(ProcessExecutor.java:1097)
        at org.zeroturnaround.exec.ProcessExecutor.execute(ProcessExecutor.java:925)
        at com.redhat.engineering.srcclr.SrcClrInvoker.execSourceClear(SrcClrInvoker.java:138)
        at com.redhat.engineering.srcclr.SCM.call(SCM.java:119)
        at com.redhat.engineering.srcclr.SCM.call(SCM.java:40)
        at picocli.CommandLine.executeUserObject(CommandLine.java:1743)
        at picocli.CommandLine.access$900(CommandLine.java:145)
        at picocli.CommandLine$RunAll.handle(CommandLine.java:2157)
        at picocli.CommandLine$RunAll.handle(CommandLine.java:2116)
        at picocli.CommandLine$AbstractParseResultHandler.handleParseResult(CommandLine.java:1928)
        at picocli.CommandLine.parseWithHandlers(CommandLine.java:2282)
        at com.redhat.engineering.srcclr.SrcClrWrapper.invokeWrapper(SrcClrWrapper.java:123)
        at com.redhat.engineering.srcclr.SrcClrWrapper.main(SrcClrWrapper.java:111)
DEBUG c.r.engineering.srcclr.SrcClrWrapper- Caught an internal exception
picocli.CommandLine$ExecutionException: Error while calling command (scanning git@github.com:strimzi/strimzi-kafka-oauth.git version null): com.redhat.engineering.srcclr.utils.InternalException: Error executing SourceClear 
        at picocli.CommandLine.executeUserObject(CommandLine.java:1752)
        at picocli.CommandLine.access$900(CommandLine.java:145)
        at picocli.CommandLine$RunAll.handle(CommandLine.java:2157)
        at picocli.CommandLine$RunAll.handle(CommandLine.java:2116)
        at picocli.CommandLine$AbstractParseResultHandler.handleParseResult(CommandLine.java:1928)
        at picocli.CommandLine.parseWithHandlers(CommandLine.java:2282)
        at com.redhat.engineering.srcclr.SrcClrWrapper.invokeWrapper(SrcClrWrapper.java:123)
        at com.redhat.engineering.srcclr.SrcClrWrapper.main(SrcClrWrapper.java:111)
Caused by: com.redhat.engineering.srcclr.utils.InternalException: Error executing SourceClear 
        at com.redhat.engineering.srcclr.SrcClrInvoker.execSourceClear(SrcClrInvoker.java:165)
        at com.redhat.engineering.srcclr.SCM.call(SCM.java:119)
        at com.redhat.engineering.srcclr.SCM.call(SCM.java:40)
        at picocli.CommandLine.executeUserObject(CommandLine.java:1743)
        ... 7 common frames omitted
Caused by: org.zeroturnaround.exec.InvalidExitValueException: Unexpected exit value: 1, allowed exit values: [0], executed command [/opt/srcclr-3.4.7/jre/bin/java, -jar, /opt/srcclr-3.4.7/srcclr-3.4.7.jar, scan, --json, --skip-collectors="SOFile", --url, git@github.com:strimzi/strimzi-kafka-oauth.git] in directory /tmp/sourceclear-invoker-7566823331311174600 with environment {GIT_BRANCH=null, GIT_URL=null, GOPATH=/tmp/sourceclear-gopath-3060457657426141852}, output was 0 bytes:

        at org.zeroturnaround.exec.InvalidExitUtil.checkExit(InvalidExitUtil.java:27)
        at org.zeroturnaround.exec.WaitForProcess.call(WaitForProcess.java:114)
        at org.zeroturnaround.exec.ProcessExecutor.waitFor(ProcessExecutor.java:1097)
        at org.zeroturnaround.exec.ProcessExecutor.execute(ProcessExecutor.java:925)
        at com.redhat.engineering.srcclr.SrcClrInvoker.execSourceClear(SrcClrInvoker.java:138)
        ... 10 common frames omitted
```

--- a/src/main/java/com/redhat/engineering/srcclr/SCM.java
+++ b/src/main/java/com/redhat/engineering/srcclr/SCM.java
@@ -74,14 +74,16 @@ public class SCM implements Callable<SourceClearResult>
         Map<String,String> env = new HashMap<>(  );
         parent.excludedEnvironment.forEach( s -> env.put( s, null ) );
 
-        if ( ! url.startsWith( "http" ) )
+        if ( url.equals( "." ))
         {
-            if ( url.equals( "." ))
-            {
-                url = Paths.get("" ).toAbsolutePath().toString();
-            }
+            url = Paths.get("" ).toAbsolutePath().toString();
+            logger.info( "Scanning local file system with {}", url );
+            args.add( url );
+        }
+        else if ( url.startsWith( "file://" ))
+        {
             String target = url.replaceFirst( "file://", "" );
-            logger.info ("Scanning local file system with {}", target);
+            logger.info( "Scanning local file system with {}", target );
             args.add( target );
         }
         else

--- a/src/main/java/com/redhat/engineering/srcclr/SrcClrInvoker.java
+++ b/src/main/java/com/redhat/engineering/srcclr/SrcClrInvoker.java
@@ -132,7 +132,7 @@ public class SrcClrInvoker
                             destroyOnExit().
                             directory( temporaryLocation.toFile() ).
                             exitValue( 0 ).
-                            redirectError( Slf4jStream.of( logger ).asDebug() ).
+                            redirectError( Slf4jStream.of( logger ).asError() ).
                             redirectOutput( Slf4jStream.of(logger).asDebug() ).
                             readOutput( true ).
                             execute().
@@ -160,7 +160,8 @@ public class SrcClrInvoker
         catch ( InvalidExitValueException e )
         {
             logger.error( e.getResult().outputUTF8() );
-            logger.error( "Invalid exit ", e );
+            logger.error( "Invalid exit: " + e.getMessage() );
+            logger.debug( "SourceClear process finished with an error ", e );
             throw new InternalException( "Error executing SourceClear ", e );
         }
         catch ( InterruptedException | TimeoutException e )

--- a/src/main/java/com/redhat/engineering/srcclr/SrcClrWrapper.java
+++ b/src/main/java/com/redhat/engineering/srcclr/SrcClrWrapper.java
@@ -60,7 +60,7 @@ public class SrcClrWrapper implements Callable<Void>
 
     private static CommandLine cl;
 
-    private final Logger logger = LoggerFactory.getLogger( SrcClrWrapper.class );
+    private static final Logger logger = LoggerFactory.getLogger( SrcClrWrapper.class );
 
     @Option( names = { "--trace" }, description = "Enable trace. Will DISABLE JSON OUTPUT" )
     private boolean trace;
@@ -145,9 +145,21 @@ public class SrcClrWrapper implements Callable<Void>
         {
             if ( e.getCause() != null && e.getCause() instanceof Exception )
             {
-                throw (Exception)e.getCause();
+                if ( e.getCause() instanceof InternalException && e.getCause().getCause() instanceof Exception)
+                {
+                    logger.debug( "Caught an internal exception", e );
+
+                    result = new SourceClearResult().setMessage( e.getCause().getCause().getMessage() );
+                }
+                else
+                {
+                    throw (Exception) e.getCause();
+                }
             }
-            throw e;
+            else
+            {
+                throw e;
+            }
         }
         return result;
     }

--- a/src/test/java/com/redhat/engineering/srcclr/internal/SourceClearInvokerTest.java
+++ b/src/test/java/com/redhat/engineering/srcclr/internal/SourceClearInvokerTest.java
@@ -63,8 +63,6 @@ public class SourceClearInvokerTest extends SCBase
         assertTrue( systemErrRule.getLog().contains( "Missing required options" ) );
     }
 
-
-
     @Test( expected = NoSuchFileException.class )
     public void runBinarySC1() throws Exception
     {

